### PR TITLE
V1.1(pt1-3): Start of types + vue tuneup + organization

### PR DESCRIPTION
**Pull request by mPaella**
Start fixing up types:

1.) Prepares for crypto pay by separating fiat/crypto types
image

2.) Trying to keep old/bad stuff separated, so we can fix in follow up prs

For now the signer interfaces are very simple, next prs will improve with more specific types/chain

image

### CODE REVIEW

File: packages/core/base/src/services/crossmintPaymentService.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
+import { FiatPaymentElementProps, PaymentElementProps } from "../types";
 import { CrossmintCheckoutEventUnion, ParamsUpdatePayload, PaymentElementSDKEvents } from "../types/events";
 import { CheckoutEvents } from "../types/events";
-import { PaymentElement } from "../types/paymentElement";
 import { getEnvironmentBaseUrl } from "../utils";
 
-export function crossmintPaymentService(props: PaymentElement) {
+// TODO: Accept both fiat and crypto
+export function crossmintPaymentService(props: FiatPaymentElementProps) {
```
Comment (guillermodoghel) [inline packages/core/base/src/services/crossmintPaymentService.ts:7]: The reference for the new component is going to be  determined by this function, or an internal redirect on 

`/sdk/paymentElement`?

File: packages/core/base/src/types/index.ts [hunk 2]
```diff
@@ -17,7 +17,18 @@ export enum baseUrls {
     dev = "http://localhost:3001",
 }
 
-export type PaymentMethod = "fiat" | "ETH" | "SOL";
+export const CryptoPaymentMethod = {
+    ETH: "ETH",
+    SOL: "SOL",
```
Comment (guillermodoghel) [inline packages/core/base/src/types/index.ts:22]: woud use lowercase to be on sync with `src/types/Currencies.ts`

File: packages/ui/vanilla-ui/src/CrossmintPaymentElement.ts [hunk 3]
```diff
@@ -28,158 +27,170 @@ const propertyDefaults: PaymentElement = {
...
-  height = 0;
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  removeEventListener = () => {};
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  removeUIEventListener = () => {};
+    height = 0;
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    removeEventListener = () => {};
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    removeUIEventListener = () => {};
```
Comment (guillermodoghel) [inline packages/ui/vanilla-ui/src/CrossmintPaymentElement.ts:85]: check linter

File: packages/core/base/src/types/PaymentElement/crypto.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+};
+
+// Signers
+export type ETHPaymentElementSigner = CommonPaymentElementSignerProps & {
+    signAndSendTransaction: (transaction: any) => Promise<any>;
+};
+
+export type SOLPaymentElementSigner = CommonPaymentElementSignerProps & {
+    signTransaction: (transaction: any) => Promise<any>;
+};
```
Comment (guillermodoghel) [inline packages/core/base/src/types/PaymentElement/crypto.ts:25]: Just to confirm, we are going to do so sign on front, broadcast on back?

File: packages/ui/vanilla-ui/src/CrossmintPaymentElement.ts (hunk 3 above)
Comment (mPaella) [reply packages/ui/vanilla-ui/src/CrossmintPaymentElement.ts:85]: this is linted correctly

File: packages/core/base/src/services/crossmintPaymentService.ts (hunk 1 above)
Comment (mPaella) [reply packages/core/base/src/services/crossmintPaymentService.ts:7]: sort of both

this function will return the correct iframe url (so, the new route only for crypto pay currently), but then we will also have different components

File: packages/core/base/src/types/PaymentElement/crypto.ts (hunk 4 above)
Comment (mPaella) [reply packages/core/base/src/types/PaymentElement/crypto.ts:25]: actually, to start i think its fine if we make this `signAndSendTransaction`

will change

File: packages/core/base/src/types/PaymentElement/crypto.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    address: string;
+};
+
+// Signers
+export type ETHPaymentElementSigner = CommonPaymentElementSignerProps & {
+    signAndSendTransaction: (transaction: any) => Promise<any>;
+};
+
+export type SOLPaymentElementSigner = CommonPaymentElementSignerProps & {
+    signTransaction: (transaction: any) => Promise<any>;
```
Comment (mPaella) [inline packages/core/base/src/types/PaymentElement/crypto.ts:24]:
```suggestion
    signAndSendTransaction: (transaction: any) => Promise<string>;
};

export type SOLPaymentElementSigner = CommonPaymentElementSignerProps & {
    signAndSendTransaction: (transaction: any) => Promise<string>;
```

**Pull request merged**